### PR TITLE
fix: Add key vault to search service - `avm/res/search/search-service`

### PR DESCRIPTION
**Pull request by mortenvpschmidt**
Add option for outputting admin keys to search service

## Pipeline Reference

| Pipeline |
| -------- |
|      [avm.res.search.search-service](https://github.com/mortenvpschmidt/bicep-registry-modules/actions/workflows/avm.res.search.search-service.yml)    |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Review by ReneHezser — DISMISSED** [review]

File: avm/ptn/security/security-center/main.json
Comment (eriqua) [inline avm/ptn/security/security-center/main.json:1]: Hey @mortenvpschmidt, it looks like unintended json files got regenerated. Please remove those changes from the PR and just keep the files related to the module you intend to contribute to.

Comment (mortenvpschmidt) [reply avm/ptn/security/security-center/main.json:1]: Hi Eriqua, thank you for catching this mistake. I should have seen that from the start!  I guess the Set-AVMModule -recurse command was started at the wrong level.

**Review by eriqua — DISMISSED** [review]
Hi @mortenvpschmidt thanks for addressing the comment and for this contribution. Be aware this falls into the same discussion opened by @Agazoth here #1934. 
I'd therefore put this PR on hold until an agreement is made there.

**Comment by mortenvpschmidt** [comment]
@microsoft-github-policy-service agree
________________________________

**Comment by Agazoth** [comment]
@AlexanderSehr thisone has been updated with the new secrets export

**Comment by AlexanderSehr** [comment]
Hey @krbar,
the PR looks good from my side and the failed job is due to an unrelated error. Any doubts?

**Comment by Agazoth** [comment]
Badge is green now. Had to change serviceShort to ssspe2 in pe test. Someone has a key vault with the shortName ssspe somewhere.

**Comment by AlexanderSehr** [comment]
> Badge is green now. Had to change serviceShort to ssspe2 in pe test. Someone has a key vault with the shortName ssspe somewhere.

Changing you namePrefix would've also worked. Regardless, @Agazoth let's do something else. Instead of renaming it from `ssspe` to `ssspe2`, let's rather do `ssspr`.
Reason being: The longer the `serviceShort` is, the higher the likelihood it will run into name-length conflicts in combination with the `namePrefix` on some resources. It's impossible to prevent for everything, but if we can help it, let's try to keep it short :)

**Comment by eriqua** [comment]
Hey @Agazoth, would you be available to address updates as per @AlexanderSehr and @krbar comments above, so that we can drive this PR to closure 💪 ?

**Comment by Agazoth** [comment]
@eriqua  and @AlexanderSehr I am so sorry - I completely missed the comment. I have updated the code and it is currently running the final test (removing resources) Looking forward to the merge!

File: avm/res/search/search-service/main.bicep [hunk 1]
```diff
@@ -530,3 +557,20 @@ type diagnosticSettingType = {
...
+
+type secretsExportConfigurationType = {
+  @description('Required. The key vault name where to store the API Admin keys generated by the modules.')
+  keyVaultResourceId: string
+
+  @description('Optional. The primaryAdminKey secret name to create.')
+  primaryAdminKey: string?
+
+  @description('Optional. The secondaryAdminKey secret name to create.')
+  secondaryAdminKey: string?
```
Comment (AlexanderSehr) [inline avm/res/search/search-service/main.bicep:569]:
```suggestion
  primaryAdminKeyName: string?

  @description('Optional. The secondaryAdminKey secret name to create.')
  secondaryAdminKeyName: string?
```

Note: Needs to be updated in other places too.

Comment (AlexanderSehr) [reply avm/res/search/search-service/main.bicep:569]: Everything else looks great. Thanks @Agazoth

Comment (Agazoth) [reply avm/res/search/search-service/main.bicep]: @AlexanderSehr  I need help. I have implemented the changes and I have run Set-AVMModule with -Recurse. Everythins should be up to date, but I keep getting this error:
{1A039F28-642C-468D-95B4-447D1A09DFAD}

Unfortunately, the info does not make me any wiser. I have rerun the Set-AVMModule cmdlet sevderal times, but there are no changes, when running it on my machine. I have checked for wierd sorting issues, but none of the parameters in the nested module have initial characters that would trigger danish sorting.

The only thing changed in the readme is this:
{9713998C-01A5-4C85-8FD8-0104DB83858F}


There seems to be missing a / in the new one and I am not sure Microsoft. part should be removed. This happened after running Set-AVMModule - I did not change the file manually.

Comment (Agazoth) [reply avm/res/search/search-service/main.bicep]: Well - that was exactly it - it even said so in the log.

I have updated the readme.mds manually.

It is really strange. Running Update-AVMModule does nothing to the ReadMes, but running Set-ModuleReadMe implements the error:
{52572748-E79A-4229-80B1-1C70DA887261}

Comment (Agazoth) [reply avm/res/search/search-service/main.bicep]: After 9 failed attempts  finally succeded in getting the REadMes in a state that the pipeline accepts. Only the pe step remains. It has been running for 52 minutes now and has been removing resources for 20+ minutes. Last time I ran the pe step successfully, it took 1 hour and 56 minutes, so I expect it to succeed in about an hour 😄

**Review by krbar — APPROVED** [review]
Then let's put a stamp on this piece of work! Thank you @Agazoth for your awesome work and also to all, who contributed!

**Pull request merged**